owtf/webapp/src/containers/TargetsPage/TargetsTable.js

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, ButtonGroup, Glyphicon } from 'react-bootstrap';
+import { BootstrapTable, TableHeaderColumn, search } from 'react-bootstrap-table';
+import {connect} from "react-redux";
+import { ClipLoader } from 'react-spinners';
+import './style.scss';
+
+
+export default class TargetsTable extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.onPageChange = this.onPageChange.bind(this);
+    this.onSearchChange = this.onSearchChange.bind(this);
+    this.onSizePerPageList = this.onSizePerPageList.bind(this);
+
+    this.state = {
+      data: this.props.targets.slice(0, 10),
+      totalDataSize: this.props.targets.length,
+      sizePerPage: 10,
+      currentPage: 1
+    };
+  }
+
+  onSearchChange(searchText, colInfos, multiColumnSearch) {
+    const currentIndex = (this.state.currentPage - 1) * this.state.sizePerPage;
+    const text = searchText.trim();
+    if (text === '') {
+      this.setState({
+        data: this.props.targets.slice(currentIndex, currentIndex + this.state.sizePerPage),
+      });
+      return;
+    }
+
+    let searchTextArray = [];
+    if (multiColumnSearch) {
+      searchTextArray = text.split(' ');
+    } else {
+      searchTextArray.push(text);
+    }
+
+    const result = this.props.targets.filter((product) => {
```

Comment:
@sharmamohit123 Can you please add some comments here? I really don't want to read whole code to understand what the variable is or functions is like **results**. Also, if you are getting data from backend in undesired format, you can change it.